### ISSUE TITLE
False positive: Service XY is private

### ISSUE THREAD

**Issue by keichinger** [phpstan/phpstan-symfony#22]
Symfony: 4.1
PhpStan: 0.10.5 

Using the following code

```php
<?php declare(strict_types=1);

// ✂️

class EntitySearchSelectorController extends AbstractController
{
    /**
     * @param Request $request
     * @param string  $entityType
     *
     * @return JsonResponse
     */
    public function doSearch (Request $request, string $entityType) : JsonResponse
    {
        switch ($entityType)
        {
            case "material":
                $model = $this->get(MaterialModel::class);
                break;

            case "manufacturer":
                $model = $this->get(ManufacturerModel::class);
                break;

            case "magazine":
                $model = $this->get(MagazineContentModel::class);
                break;

            default:
                throw $this->createNotFoundException(\sprintf(
                    "Unsupported entity type „%s”.",
                    $entityType
                ));
        }
        // ✂️
    }

    /**
     * @inheritdoc
     */
    public static function getSubscribedServices () : array
    {
        return \array_replace(parent::getSubscribedServices(), [
            MagazineContentModel::class,
            ManufacturerModel::class,
            MaterialModel::class,
        ]);
    }
}
```
I'll get an error where PhpStan complains that my services are `private` - which is the default for Symfony services. In this example I've marked two of my three services explicitly as `public`. However, it shouldn't be needed in this case. The code is working perfectly fine in both `dev` and `prod` environment. It's just PhpStan that is complaining.

In this project we've adopted the bundle-less approach, as recommended by Symfony.

![image](https://user-images.githubusercontent.com/439899/49211004-d2c10100-f3be-11e8-9367-1ee381946827.png)

The `config/services.yaml` is also nothing fancy and looks like this:

```yaml
services:
    _defaults:
        autowire: true
        autoconfigure: true
        public: false

    App\:
        resource: '../src/*'
        exclude: '../src/{DependencyInjection,Entity,Migrations,Tests,Kernel.php}'

    # Uncommented for the sake of reproducing/reporting this issue
    #App\Model\Magazine\MagazineContentModel:
    #    # required to keep PhpStan silent
    #    public: true

    App\Model\ManufacturerModel:
        # required to keep PhpStan silent
        public: true

    App\Model\Material\MaterialModel:
        # required to keep PhpStan silent
        public: true
```

**Comment by apfelbox** [comment]
The issue seems to be that services that are explicitly requested via the `ServiceSubscriberInterface` are not marked as accessible / public.

**Comment by lookyman** [comment]
Thank you, I will look into it.

**Comment by lookyman** [comment]
Fixed in  c340de8.

**Comment by keichinger** [comment]
Amazing! Thank you very much @lookyman for your hard work!
